### ISSUE TITLE
Unhandled rejection Error when refresh token is invalid

### ISSUE THREAD

**Issue by TuomasTammi** [Asana/node-asana#148]
When refresh token is invalid there is an unhandled rejection error. 

For example if you run the code below by supplying valid credentials and a valid refresh token it performs correctly. Output is `Success!`.

```
var asana =  require('asana')
    .Client
    .create({
        clientId: 'VALID_ID',
        clientSecret: 'VALID_SECRET',
        redirectUri: 'VALID_REDIRECT_URI'
    })
    .useOauth({
        credentials: {
            access_token: 'INVALID',
            refresh_token: "VALID_OR_INVALID"
        }
    });

return asana.users.me()

.then(function () {
    console.log('Success!');
})

.catch(function(err) {
    console.log('Error!');
});
```

If you supply an invalid refresh token, then instead of getting the expected output `Error!`, that catch block is never reached. There is only this text in the console

```
Unhandled rejection Error: {
  "error": "invalid_grant",
  "error_uri": "https://asana.com/developers/documentation/getting-started/authentication",
  "error_description": "The `refresh_token` provided was invalid."
}
    ... stack trace omitted ...
```

Using incorrect credentials has similar results.

**Comment by praecipula** [comment]
➤ Jeff Schneider commented:

Merged pull request.

**Comment by theaeolianmachine** [comment]
Closed by #149
